### ISSUE TITLE
With doesn't work

### ISSUE THREAD

**Issue by benmoss** [thessem/zap-prettyconsole#1]
I was trying to use your project's logger and ran into this issue with zap's `With` method. It seems like the prettyconsole encoder doesn't print the extra fields. 

```go
package main

import (
	prettyconsole "github.com/thessem/zap-prettyconsole"
	"go.uber.org/zap"
)

func main() {
	pretty := zap.Must(prettyconsole.NewConfig().Build())
	pretty = pretty.With(zap.String("foo", "foo"))
	pretty.Debug("wtf", zap.String("bar", "bar"))

	dev := zap.Must(zap.NewDevelopmentConfig().Build())
	dev = dev.With(zap.String("foo", "foo"))
	dev.Debug("wtf", zap.String("baz", "bar"))
}
```

> $ go run ./
>  12:02PM DBG > wtf bar=bar
> 2022-12-01T12:02:39.113-0500    DEBUG   foo/main.go:15  wtf     {"foo": "foo", "baz": "bar"}

https://go.dev/play/p/IVPYVYRduhC

**Comment by thessem** [comment]
Thanks for taking the time to report this. An oversight on my behalf, should be fixed with v0.3.0. Your go playground example seems to be working properly. The playground is not doing well with the console colour commands though!

Unfortunately my desire to have alphabetically sorted fields has had a bit of an effect on the benchmarks. The way the JSON encoder handles this is fairly elegant, but since I want to re-sort the fields whenever something is logged I don't think there's a more performant way on my side. I realised I was in trouble when I had to encode new fields without even knowing what the log level is yet 😅
